mne/viz/evoked.py

```diff
@@ -2127,6 +2133,11 @@ def plot_compare_evokeds(evokeds, picks=None, gfp=False, colors=None,
     evokeds, colors = _format_evokeds_colors(evokeds, cmap, colors)
     conditions = sorted(list(evokeds.keys()))
 
+    if ci and any(len(evokeds_) < 2 for evokeds_ in evokeds.values()):
+        if not isinstance(ci, bool):
+            warn("Only one Evoked object per condition. Not plotting the CI.")
+        ci = False
```

Comment:
once you switched to 'auto' option isolate the logic of setting ci with a private function.
Something like `_check_ci`